src/transfer.rs

```diff
@@ -395,21 +394,20 @@ where
             }
         } else {
             // Create a stream of tasks for uploading the blocks
-            let file = Arc::new(File::open(source)?);
             let offset = Arc::new(AtomicU64::new(0));
             let mut stream = stream::iter(0..info.num_blocks)
                 .map(|_| {
+                    let source= source.to_path_buf();
```

Comment:
Unsure why rustfmt gave up on this one, but:
```suggestion
                    let source = source.to_path_buf();
```